### PR TITLE
allow clear text username:password credentials in dockercfg

### PR DESCRIPTION
**Pull request by Vudumen**

### CODE REVIEW

**Comment by cpuguy83** [comment]
Is there a particular reason we would want to allow clear text here?  Not that base64 is the best thing either but...

**Comment by Vudumen** [comment]
As we use Docker it would be much better for us to be able to easily see (and edit) the content of this attribute in the config file. In this way I've implemented it should not break anything.

File: registry/auth_test.go [hunk 1]
```diff
@@ -24,6 +24,14 @@ func TestEncodeAuth(t *testing.T) {
 	if authStr != "a2VuOnRlc3Q=" {
 		t.Fatal("AuthString encoding isn't correct.")
 	}
+	username0, password0, _ := decodeAuth("dXNlcm5hbWU6cGFzc3dvcmQ=")
+	if username0 != "username" || password0 != "password" {
```
Comment (ahmetalpbalkan) [inline registry/auth_test.go]: check for err?

Comment (Vudumen) [inline registry/auth_test.go]: Thanks, I added the tests.

Comment (ahmetalpbalkan) [inline registry/auth_test.go]: Oh I was just saying if err!= nil t.Fatal(err). These  cases you newly added because of my comment seems unnecessary. :-)

**Comment by ahmetalpbalkan** [comment]
@vudumen I believe you need to squash your comments into one so that it is stable, atomic and revert-able in case someone needs to undo. :-)

**Comment by Vudumen** [comment]
I squashed my commits and added test for the returned error vars.

**Comment by icecrime** [comment]
I think it's easy enough to use [`base64`](http://linux.die.net/man/1/base64) to decode/edit/encode when editing in clear is desirable.

Sorry, I'm -1 on design for this, but I'm not closing yet so feel free (or another maintainer) to prove me wrong!

**Comment by dmp42** [comment]
-1 as well. Base64 is not encryption, but at least it lets you avoid messy encoding issues.
Either way, we are likely to move away from plain text credentials being stored locally.

**Comment by tiborvass** [comment]
I personally don't think base64 or clear text is any different, and theoretically we could base64 encode/decode on the fly.

I think the strongest case against this proposal is that adding support for yet another format (content contains a colon `:`) is not really desired given that we'd like to move away from plain text like dmp42 said.

@dmp42 maybe you could detail here or provide a link to what the new credentials storage system is intended to be?

@Vudumen let us know what you think. Thanks!

**Comment by Vudumen** [comment]
@tiborvass: I think that as I wrote it would be useful for us. We are working on deploying Docker to our infrastructure and development environment and this would make our life much easier when we have to debug things. We already met other bugs in Docker & Docker registry and when we tried investigating them we found many times that it was authentication. Of course it's a special case where we have our own SSO implementation and a few other suff but we want to integrate Docker into it.
@icecrime: I agree with you that base64 encoding/decoding is easy but I have to thing about other people. Let's say a typical problem is when they use echo "a:b" | base64 and they forget the -n switch to echo.
@dmp42: I agree with you too but I would go further and say that base64 == cleartext at least from security view. There are other software products - some of them older or bigger - which uses cleartext credentials too. Let's say .my.cnf for Mysql contains cleartext info but all of us know encrypted filesystems and chmod... Maybe if you check the permissions of the file (just like as sshd does with it's keys) and warn the user if it's world/group readable.

**Comment by icecrime** [comment]
@Vudumen I think this is not sufficiently of a common use case to justify a patch. If it's easy enough *for you* to workaround it then we're good: I don't see a good reason to make it easier for everyone.

We nevertheless appreciate the time and effort you put in the PR, sorry I'm closing this.

**Pull request closed without merging**